routes/words.py

```diff
@@ -31,30 +30,94 @@
 
 from settings import changedlocale
 
+from tokenizer import TOK
+
 from reynir.bindb import BIN_Db
+from reynir.bintokenizer import tokenize
 
 from db import SessionContext, desc
-from db.models import Article, Root, Location, ArticleTopic, Topic
+from db.models import Article, Word, Root
 from db.queries import WordFrequencyQuery
 
 
 @routes.route("/words")
 def words():
     """ Handler for word frequency page. """
-    return render_template("words.html", title="Orð")
+    return render_template("words/freq.html", title="Orð")
+
+
+@routes.route("/words_trends")
+def words_trends():
+    """ Handler for word trends page. """
+    return render_template("words/trends.html", title="Orð")
+
+
+# Word categories permitted in word frequency search
+_VALID_WCATS = frozenset(
+    ("kk", "kvk", "hk", "lo", "so", "person_kk", "person_kvk", "entity", "??")
+)
+
+# Human-readable descriptions of word categories
+CAT_UNKNOWN = "??"
+CAT_DESC = {
+    "kk": "kk. no.",
+    "kvk": "kvk. no.",
+    "hk": "hk. no.",
+    "lo": "lo.",
+    "so": "so.",
+    "person_kk": "kk. manneskja",
+    "person_kvk": "kvk. manneskja",
+    "entity": "fyrirbæri",
+    CAT_UNKNOWN: "óþekkt",
+}
+
+# Tokens that should not be ignored
+_VALID_TOKENS = frozenset((TOK.WORD, TOK.PERSON, TOK.ENTITY))
 
+# Max number of words to request frequency data for simultaneously
+_MAX_NUM_WORDS = 6
 
+# Max period length before grouping chart stats by week
+_SHOW_WEEKS_CUTOFF = 90  # days
+
+# Word frequency chart line colors (Miðeind colors)
 _LINE_COLORS = frozenset(
     ("#006eff", "#eb3732", "#00b450", "#ffb400", "#4600c8", "#f0f")
 )
 
 
+def _str2words(wstr, separate_on_whitespace=False):
+    """ Parse string of the form 'w1:cat1, w2:cat2, ...' into a list
+        of word/category tuples. """
+    if not wstr:
+        return None
+    ws = wstr.strip()
+    if not ws:
+        return None
+    words = [w.strip() for w in ws.split(",")]
+    if separate_on_whitespace:
+        words = " ".join(words).split()
+    return [
+        w.split(":") if len(w.split(":")) == 2 else (w, CAT_UNKNOWN) for w in words
+    ][:_MAX_NUM_WORDS]
+
+
+def _words2str(words):
+    """ Create comma-separated string from (word,cat) tuple list,
+        e.g. "[(a,b),(c,d)] -> "a:b, c:d". """
+    return ", ".join([":".join(w[:2]) if len(w) >= 2 else (w, None) for w in words])
```

Comment:
Should probably be `", ".join([":".join(w[:2]) if len(w) >= 2 else w[0] for w in words])`